src/sleep/sleep.c

```diff
@@ -284,6 +284,11 @@ static int execute_s2h(const SleepConfig *sleep_config) {
 
                 before_timestamp = now(CLOCK_BOOTTIME);
 
+                previous_discharge_rate = get_battery_discharge_rate();
+                if (previous_discharge_rate > 0)
+                        suspend_interval = ((last_capacity / previous_discharge_rate) * 60 - 30) * USEC_PER_MINUTE;
+                else if (previous_discharge_rate != -ENOENT)
+                        log_error_errno(previous_discharge_rate, "Error fetching battery discharge rate, ignoring: %m");
```

Comment:
We ignore an error here and log at error but ignore an error below as well and log at warning? I'd say we either log at error always or log at warning always, instead of mixing the two.

Reply:
updated to log as error